dotnet-algorand-sdk/V2/Indexer/SearchApi.cs

```diff
@@ -74,7 +74,7 @@ public partial interface ISearchApi
         /// <param name="asset_id">Asset ID</param>
         /// <returns>(empty)</returns>
         /// <exception cref="ApiException">A server side error occurred.</exception>
-        System.Threading.Tasks.Task<Response3> AssetsAsync(bool? include_all, int? limit, string next, string creator, string name, string unit, int? asset_id);
+        System.Threading.Tasks.Task<Response3> AssetsAsync(bool? include_all = null, int? limit = null, string next = null, string creator = null, string name = null, string unit = null, ulong? asset_id = null);
```

Comment:
This follows the same pattern as the ApplicationsAsync, but this one is a bit more valid. For instance, we could look up all assets with a unitName of "Hello". We do not want to restrict this to certain asset ids. We could certainly discuss parameter order, but asset id is not actually the primary entry here. If you were looking to get a single asset you should use the LookupApi

Reply:
First, thanks very much guys for this. My view is from the maintenance point of view. For the time being we need to assume updates to the OpenAPI specs on the Algorand side can be breaking/significantly large, and that generation of APIs and Models will probably be automated. Right now the situation is that we can regenerate the APIs using NSwag and then replace the ints with ulongs, perhaps manually.  The plan is that in the near future I will cooperate with Algorand on updating their 'internal' codegen tool, which they use to generate the core clients. This will effectively, though perhaps not officially, make the .NET SDK a core sdk too. One interesting thing you may have noticed too is that the OpenAPI specs do not declare that every integer should be ulong. For example, 'number of extra pages' is a default integer, usually interpreted as 32 bit. Algorand's internal codegen is also used to generate the server part of go-algorand client. It seems at the moment that the codegen tool is still young, and only loosely interprets the spec, roughly defaulting to uint64 for any integer. I don't think this is something that will remain the case too. So, one way or another, either by temporary NSwag based maintenance, or later by Algorand Codegen, the API "surface" and its types should involve I think as little manual intervention as possible, so that there's less chance of having to reapply those changes each time the OpenAPI specs change. The alternative is that any updates to the APIs involve manually diffing the specs from version to version and manually coding up the changes. 

I think therefore that for functions that offer searching for concrete asset ids or other specific cases, I'd perhaps suggest a PR with new utilities, eg SearchUtilities, wrapping the SearchApi